lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -3295,15 +3300,17 @@ CheckedCastKind TypeChecker::typeCheckCheckedCast(Type fromType,
                                          BridgingCoercion);
         break;
 
+      case CheckedCastKind::Unresolved:
+        if (contextKind == CheckedCastContextKind::Coercion)
+          return failed();
+        LLVM_FALLTHROUGH;
```

Comment:
Could you also add a comment here? It can be something brief like "Handled the same as in checkElementCast; see comment there for rationale."